harness/determined/exec/harness.py

```diff
@@ -29,6 +31,10 @@ def main(train_entrypoint: str) -> int:
     # TODO: refactor data_layer, and profiling to to not use the cli_cert.
     certs.cli_cert = certs.default_load(info.master_url)
 
+    trial_class = load.trial_class_from_entrypoint(train_entrypoint)
+    if issubclass(trial_class, PyTorchTrial):
+        create_pytorch_trial_controller_and_context(info)
```

Comment:
I would expect this this if statement to have a `return` in it.  Let if statement this be the hook that separates the legacy codepath from the legacy-to-modern conversion path, and probably don't mix them.

Reply:
side note, I like passing in `info`, and I would imagine that there's nothing else you would need to pass.

Reply:
side side note: maybe include the word `legacy` or `compatibility` or something in the name of the `create_pytorch_trial_controller_and_context` function?

It would be nice if it were more obvious to readers that this is a compatibility layer from old mechanisms to new mechanisms.